lib/Bio/EnsEMBL/DataCheck/Checks/CheckHomology.pm

```diff
@@ -44,16 +44,22 @@ sub tests {
   is_one_to_many($dbc, "homology_member", "homology_id", $desc_1);
   ### Hoping for a better idea than this query (below)
   my $hideous_sql = q/
-    SELECT hm1.gene_member_id gene_member_id1, hm2.gene_member_id gene_member_id2, COUNT(*) num, 
-      GROUP_CONCAT(h1.description order by h1.description) descs 
-        FROM homology h1 
-          CROSS JOIN homology_member hm1 
-            USING (homology_id)
-          CROSS JOIN homology_member hm2 
-            USING (homology_id)
-    WHERE hm1.gene_member_id < hm2.gene_member_id
-      GROUP BY hm1.gene_member_id, hm2.gene_member_id 
-        HAVING COUNT(*) > 1
+    SELECT
+      hm1.gene_member_id gene_member_id1,
+      hm2.gene_member_id gene_member_id2,
+      COUNT(*) num,
+      GROUP_CONCAT(h1.description
+          ORDER BY h1.description) descs
+    FROM
+      homology h1
+          CROSS JOIN
+      homology_member hm1 USING (homology_id)
```

Comment:
imho, since `CROSS JOIN` is filtered on `homology_id` it will produce the exact same result as an inner join, I am not sure if that has an impact on performance, (depend on how mysql optimize the resulting data access) but CROSS JOIN is generally a "I want to kill my server" kind of thing.

Reply:
This is a horrible "legacy" style query that was simply taken from the java healthcheck version, and now altered. I honestly strongly dislike the entire query, but have been unable to improve upon it. We do need to get all the possible lines in the table because of the nature of the way the relationships are defined in the `homology` and `homology_member` tables - really the best we can hope for is an improved schema design/model to store these homology relationships :crossed_fingers:

Reply:
As I said the CROSS being filtered on `homology_id` the result is NOT all the possible lines though, only the matched ones. I was just concerned about MySQL doing the CROSS `then` filtering, or automatically move to a more simple `INNER`.

Reply:
I'll wait for @twalsh-ebi approval before merging anyway

Reply:
@marcoooo - Thank you and I think what you say makes sense. I will write a ticket for improving the query on our side based on what you have said for another release, at the moment I don't think we have the time to test another change

Reply:
NP.